<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- arguments -->
    <arg name="gamma_1" default="0.3" />
    <arg name="gamma_2" default="0.3" />
    <arg name="h" default="1.0" />

    <arg name="gui" default="true" />
    <arg name="world_name" default="$(find gazebo_ros)/worlds/empty.world" />
    <arg name="front_laser" default="false" />

    <!-- spawn Jackal robot -->
    <include file="$(find jackal_gazebo)/launch/empty_world.launch">
        <arg name="gui" value="$(arg gui)" />
        <arg name="world_name" value="$(arg world_name)" />
        <arg name="front_laser" value="$(arg front_laser)" />
    </include>

    <!-- run controller -->
    <node pkg="robot_sim_cpp" type="wheeled_mobile_robot_pose_control_node"
            name="wheeled_mobile_robot_pose_control_node" respawn="false" output="screen">
        <param name="wheeled_mobile_robot_pose_control/gamma_1" type="double" value="$(arg gamma_1)" />
        <param name="wheeled_mobile_robot_pose_control/gamma_2" type="double" value="$(arg gamma_2)" />
        <param name="wheeled_mobile_robot_pose_control/h" type="double" value="$(arg h)" />
    </node>
</launch>